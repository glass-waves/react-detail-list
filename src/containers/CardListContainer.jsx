import React, { Component } from 'react';
import CardList from '../components/card/CardList';
import { getMagicCards } from '../services/getMagicCards';

export default class CardListContainer extends Component {
    state = {
        cardList: [],
        loading: true,
    };

    async componentDidMount() {
        const cards = await getMagicCards();
        this.setState({
            cardList: cards,
            loading: false
        });

    }
    render() {
        return (
            <div>
                {this.state.loading && <h1>Loading</h1>}
                <CardList cardList={this.state.cardList}/>
            </div>
        );
    }
}
