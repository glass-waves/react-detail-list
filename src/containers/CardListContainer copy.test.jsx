import React from 'react';
import { render, screen } from '@testing-library/react';
import CardListContainer from './CardListContainer';

describe('CardListContainer', () => {
    it('renders a list of magic cards to the page', async () => {
        render(<CardListContainer />);
        screen.getByText('Loading');
        const ul = await screen.findAllByRole('list', { name: 'cards' });
        expect(ul).toMatchSnapshot();
    });
});
