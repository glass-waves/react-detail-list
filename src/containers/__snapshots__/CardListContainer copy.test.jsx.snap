// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListContainer renders a list of magic cards to the page 1`] = `
Array [
  <ul
    aria-label="cards"
  />,
]
`;
